### PR TITLE
Clean up those error parsers

### PR DESCRIPTION
**Pull request by monkeydl**
This pull addresses the following:

**Rendering of Errors message**
- Removing status code from the render message body, we prefer using HTTP status code in headers for a more standardized FE experience)
- Add support for additional data type (hash, hash array, string & string array)

**Refactoring the main renderer class**
- Move type checks into its respective renderer
- Re-write record renderer and error renderer :yay:
- Remove redundant checks in and outside data parsers and error parsers

**Update tests**
- Heaps of redundant tests that can be grouped into one place, they are now grouped
- Add additional test scenarios for additional data types

**Update version 0.2.1**

### CODE REVIEW

File: lib/google_json_response/record_parsers/base.rb [hunk 1]
```diff
@@ -37,7 +37,7 @@ def pagination_data
       def sort_data
         return { sort: options[:sort] } if options[:sort]
         return { sort: options[:sorts].join(',') } if options[:sorts].is_a?(Array)
-        { sort: nil }
+        {}
```
Comment (monkeydl) [inline lib/google_json_response/record_parsers/base.rb]: cc @huy120486 yours

**Comment by monkeydl** [comment]
WIP for the time being, because i'm still moving those tests into the google_json_response_spec file. But for now you can look at the logik 👍

File: README.md [hunk 2]
```diff
@@ -101,14 +100,13 @@ We use the custom error class in our purchasing service.
 Now we want to render the error at the application layer (Rails controller for example)
 ```ruby
   if !service.success?
-    return GoogleJsonResponse.render_error(service.errors, code: 400).to_json
+    render json: GoogleJsonResponse.render_error(service.errors).to_json, status: 500
```
Comment (dangnm) [inline README.md]: @monkeydl where is the method "render"?

**Comment by monkeydl** [comment]
Ready for review!

File: README.md [hunk 3]
```diff
@@ -185,13 +191,12 @@ The result will be like this
 #### Scenario 4: Render a generic error message
 Sometimes we will need to render a simple error message at application layer
 ```ruby
-  GoogleJsonResponse.render_error("You can't access this page", code: '401').to_json
+  GoogleJsonResponse.render_error("You can't access this page").to_json
```
Comment (ehhuythai) [inline README.md]: why don't the `render_error` returns a json?

Comment (monkeydl) [reply README.md]: it's ok to return whatever I think. A hash just happens to be more flexible than a JSON :troll:

File: lib/google_json_response/error_parsers/parse_errors.rb [hunk 4]
```diff
@@ -1,76 +0,0 @@
...
-      def initialize(data, options = {})
-        @options = options
-        @code = @options[:code]
-        @data = data
-        @errors = []
-      end
-
-      def call
-        if @data.is_a?(::Array)
-          @parsed_data = parse_error_array(@data)
```
Comment (dangnm) [inline lib/google_json_response/error_parsers/parse_errors.rb]: @longnh92 it's here! 😢

**Pull request merged**